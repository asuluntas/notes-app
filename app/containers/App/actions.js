import {
  LOAD_NOTES,
  LOAD_NOTES_SUCCESS,
  LOAD_NOTES_ERROR,
} from '../Notes/constants';

import {
  ADD_NOTE,
  ADD_NOTE_SUCCESS,
  ADD_NOTE_ERROR,
} from '../HomePage/constants';

export function loadNotes() {
  return {
    type: LOAD_NOTES,
  };
}

export function notesLoaded(notes, note) {
  return {
    type: LOAD_NOTES_SUCCESS,
    notes,
    note,
  };
}

export function notesLoadingError(error) {
  return {
    type: LOAD_NOTES_ERROR,
    error,
  };
}

export function addNote(note) {
  return {
    type: ADD_NOTE,
    note,
  };
}

export function noteAdded(note) {
  return {
    type: ADD_NOTE_SUCCESS,
    note,
  };
}

export function noteAddingError(error) {
  return {
    type: ADD_NOTE_ERROR,
    error,
  };
}
