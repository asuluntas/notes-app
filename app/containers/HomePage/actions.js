import { CHANGE_NOTE } from './constants';

export function changeNote(note) {
  return {
    type: CHANGE_NOTE,
    note,
  };
}
