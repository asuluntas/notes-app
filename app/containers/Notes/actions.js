import { LOAD_NOTES, LOAD_NOTES_SUCCESS, LOAD_NOTES_ERROR } from './constants';

export function loadNotes() {
  return {
    type: LOAD_NOTES,
  };
}

export function notesLoaded(notes) {
  return {
    type: LOAD_NOTES_SUCCESS,
    notes,
  };
}

export function notesLoadingError(error) {
  return {
    type: LOAD_NOTES_ERROR,
    error,
  };
}
