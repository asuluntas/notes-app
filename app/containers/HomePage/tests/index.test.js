import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { HomePage, mapDispatchToProps } from '../index';
import { changeNote, addNote } from '../actions';
import configureStore from '../../../configureStore';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage
            note=""
            loadingAddNote={false}
            addNoteError={false}
            recentlyAddedNote={null}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should not call onSubmitForm if note is an empty string', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage onChangeNote={() => {}} onSubmitForm={submitSpy} />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  it('should not call onSubmitForm if note is null', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage note="" onChangeNote={() => {}} onSubmitForm={submitSpy} />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('onChangeNote', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeNote).toBeDefined();
      });

      it('should dispatch changeNote when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const note = 'a test note';
        result.onChangeNote({ target: { value: note } });
        expect(dispatch).toHaveBeenCalledWith(changeNote(note));
      });
    });

    describe('onSubmitForm', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onSubmitForm).toBeDefined();
      });

      it('should dispatch addNote when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const note = 'a test note';
        result.onSubmitForm({ target: { value: note } });
        expect(dispatch).toHaveBeenCalledWith(addNote(note));
      });

      it('should preventDefault if called with event', () => {
        const preventDefault = jest.fn();
        const result = mapDispatchToProps(() => {});
        const evt = { preventDefault };
        result.onSubmitForm(evt);
        expect(preventDefault).toHaveBeenCalledWith();
      });
    });
  });
});
