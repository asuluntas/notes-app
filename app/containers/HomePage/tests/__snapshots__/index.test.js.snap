// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<article>
  <div>
    <section
      class="Section-sc-1eulrsm-0 eTEsZl"
    >
      <h2
        class="H2-egphza-0 HsCXl"
      >
        <span>
          Write a Note!
        </span>
      </h2>
      <form
        class="Form-dmhz4x-0 kaViEG"
      >
        <label>
          <span>
            Your Note here 
          </span>
          <input
            class="Input-sc-19rkjmi-0 eXlGFi"
            id="note"
            placeholder="write something..."
            required=""
            type="text"
            value=""
          />
        </label>
      </form>
    </section>
  </div>
</article>
`;
